sys/disp_dev/disp_dev.c

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2020 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @brief       Helper functions for generic API of display device
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ *
+ * @}
+ */
+
+#include <assert.h>
+#include <inttypes.h>
+
+#include "disp_dev.h"
+
+void disp_dev_map(disp_dev_t *dev,
```

Comment:
Shouldn't these be "static inline" functions? Or is that premature optimization?

And, if there are reasonable defaults, maybe do sth like
```
if (!dev->driver->width) return $DEFAULT;
```

... or return an error?

That way, not all drivers have to provide a function for everything but can only define the actually implemented ones.

Reply:
> Shouldn't these be "static inline" functions? Or is that premature optimization?

This is a public API, it's meant to used from user code.

> or return an error?

How about adding an assert in the wrapper functions ?

```c
assert(dev->driver->map);
```

Reply:
> This is a public API, it's meant to used from user code.

Yeah, still, a function prologue /epilogue is longer than two pointer dereferences, no? (goes to try quickly)

> How about adding an assert in the wrapper functions ?

That would mean all disp_devs have to implement all functions. We could do that.
But if there's a use-case for not doing that (e.g., with sane defaults or error values), not defining a function (function pointer is NULL) could mean either "use default" or "not supported / implemented".

Reply:
> Yeah, still, a function prologue /epilogue is longer than two pointer dereferences, no? (goes to try quickly)

seems to be the same flash usage, so let'S keep it as is.

Reply:
> That would mean all disp_devs have to implement all functions. We could do that.

Let's go with that for now.